### PR TITLE
Prefer encrypt setting

### PR DESCRIPTION
**Pull request by Valodim**
Adds a setting that enables Autocrypt's `prefer-encrypt=mutual`. The logic is trivial since it was already there, the hard part is the wording I guess. I'm ok with this for a first attempt, but there is probably room for improvement. In a way, the goal is to get the user *not* to just blindly enable this setting, like any variant of "Prefer encryption" or "Encrypt by default" would do.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java [hunk 1]
```diff
@@ -711,6 +712,9 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             });
 
             cryptoMenu.setOnPreferenceClickListener(null);
+
+            autocryptPreferEncryptMutual = (CheckBoxPreference) findPreference(PREFERENCE_NOTIFICATION_OPENS_UNREAD);
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java]: This is wrong.

File: k9mail/src/main/res/xml/account_settings_preferences.xml [hunk 2]
```diff
@@ -479,6 +479,12 @@
             android:key="crypto_key"
             android:title="@string/account_settings_crypto_key" />
 
+        <CheckBoxPreference
+            android:key="autocrypt_prefer_encrypt"
```
Comment (philipwhiuk) [inline k9mail/src/main/res/xml/account_settings_preferences.xml]: I'd rather setting was either completely crypto agnostic or not. i.e. just "prefer_encrypt"

File: k9mail/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1280,4 +1280,7 @@ Please submit bug reports, contribute new features and ask questions at
     <string name="openpgp_enabled_error_back">Back</string>
     <string name="openpgp_enabled_error_disable">Disable Encryption</string>
     <string name="openpgp_encryption">OpenPGP Encryption</string>
+    <string name="account_settings_crypto_prefer_encrypt">Encryption preference</string>
+    <string name="account_settings_crypto_prefer_encrypt_on">By default when replying to encrypted, or if all recipients also checked this option</string>
```
Comment (philipwhiuk) [inline k9mail/src/main/res/values/strings.xml]: `also checked this option` -> `also would like encryption`

File: k9mail/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1280,4 +1280,7 @@ Please submit bug reports, contribute new features and ask questions at
...
+    <string name="account_settings_crypto_prefer_encrypt_off">By default, when replying to encrypted messages</string>
```
Comment (philipwhiuk) [inline k9mail/src/main/res/values/strings.xml]: 'when' -> 'only when'

File: k9mail/src/main/res/xml/account_settings_preferences.xml (hunk 2 above)
Comment (Valodim) [reply k9mail/src/main/res/xml/account_settings_preferences.xml]: This is not a generic setting - unlike any other, the semantics are precisely defined by autocrypt. Hence the name

File: k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java (hunk 1 above)
Comment (Valodim) [reply k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java]: Derp.

File: k9mail/src/main/res/values/strings.xml (hunk 3 above)
Comment (Valodim) [reply k9mail/src/main/res/values/strings.xml]: This is a really tough one. Everyone generally "likes" or "prefers" encryption over not having that. But it's quite specifically checking this particular option (and only in an Autocrypt-capable client) that triggers it... hmm. I wonder if we should have a specific name (that can be exposed to users) for this setting it in the spec.

**Comment by philipwhiuk** [comment]
Do you need a preferences upgrade process.

Also, do we need to do anything to make sure this preference is exported in the XML (can't remember if there's a code change needed there or not).

**Comment by Valodim** [comment]
Not happy yet with the captions

**Pull request merged**